src/main/java/mods/railcraft/data/worldgen/RailcraftConfigurableOre.java

```diff
@@ -0,0 +1,177 @@
+  private List<Pair<RuleTest, ResourceLocation>> oreBlockTypes;
+  private int count;
+  private int maxGroupSize;
+  private int minY;
+  private int maxY;
+
+  /* === CONFIGURATION VALUES === */
+  private EnumValue<PLACEMENT_TYPE> placementTypeConfig;
+  private IntValue countConfig;
+  private IntValue maxGroupSizeConfig;
+  private IntValue minYConfig;
+  private IntValue maxYConfig;
+  private ConfigValue<List<? extends String>> biomeIgnoreList;
+
+  /**
+   * Creates a new configurable ore spawn
+   * Will only take config changes during game restarts (or maybe during data reload (biomes etc) as we register there.)
+   * @param id - The id used in config, PlacedFeature, and ConfiguredFeature
+   * @param oreBlockTypes - A list containing another list with its variants.
+   * @param placementType
+   * @param count
+   * @param maxVeinCount
+   * @param minY
+   * @param maxY
+   */
+  public RailcraftConfigurableOre(String id, List<Pair<RuleTest, ResourceLocation>> oreBlockTypes,
+      PLACEMENT_TYPE placementType, int count, int maxGroupSize, // pls confirm if its max vein count, count one is chunk max i think
+      int minY, int maxY) {
+
+    this.oreBlockTypes = oreBlockTypes;
+    this.placementType = placementType;
+    this.count = count;
+    this.maxGroupSize = maxGroupSize;
+    this.minY = minY;
+    this.maxY = maxY;
+    this.resourceLocation = new ResourceLocation(Railcraft.ID, id);
+  }
+
+  /**
+   * Generate the config spec for this ore generator.
+   */
+  public void generateOreConfig(Builder builder) {
+    builder.comment("Ore generation for " + resourceLocation.getPath())
+      .push(resourceLocation.getPath());
+
+    this.placementTypeConfig = builder
+        .comment("What placing system should this ore generator use? RARE (average) or COMMON (max count in chunk)")
+        .defineEnum("placementType", this.placementType);
+
+    this.countConfig = builder
+        .comment("Maximum amount of ores that can spawn in a vein? (unconfirmed)")
+        .defineInRange("count", this.count, 0, 64);
+
+    this.maxGroupSizeConfig = builder
+        .comment("Maximum amount of ores that can spawn in a clump. If RARE, it will be the average.")
+        .defineInRange("maxGroupSize", this.maxGroupSize, 0, 256);
+
+    this.minYConfig = builder
+        .comment("Minimum y level spawn")
+        .defineInRange("minY", this.minY, -32, 319);
+
+    this.maxYConfig = builder
+        .comment("Maximum y level spawn")
+        .defineInRange("maxY", this.maxY, -32, 319);
+
+    this.biomeIgnoreList = builder
+        .comment("Biome ignore list. Must be a valid resource path (eg: 'minecraft:nether').")
+        .defineList("biomeIgnoreList", List.<String>of(),
+            obj -> ResourceLocation.isValidResourceLocation(obj.toString()));
+    builder.pop();
+  }
+
+  /**
+   * Hook for reloading ore configs.
+   */
+  public void handleConfigSpecLoad() {
+    this.placementType = this.placementTypeConfig.get();
```

Comment:
Why are you copying these values? (Why can't you just reference the config directly)